db.go

```diff
@@ -0,0 +1,83 @@
+package notmuch
+
+// #cgo LDFLAGS: -lnotmuch
+// #include <stdlib.h>
+// #include <notmuch.h>
+import "C"
+
+import (
+	"runtime"
+	"unsafe"
+)
+
+const (
+	// DBReadOnly is the mode for opening the database in read only.
+	DBReadOnly = C.NOTMUCH_DATABASE_MODE_READ_ONLY
+
+	// DBReadWrite is the mode for opening the database in read write.
+	DBReadWrite = C.NOTMUCH_DATABASE_MODE_READ_WRITE
+)
+
+type (
+	// DBMode is the mode of the database opening, DBReadOnly or DBReadWrite
+	DBMode C.notmuch_database_mode_t
+
+	// DB represents a notmuch database.
+	DB struct {
+		cptr unsafe.Pointer
+	}
+
+	status C.notmuch_status_t
+)
+
+func (db *DB) toC() *C.notmuch_database_t {
+	return (*C.notmuch_database_t)(db.cptr)
+}
+
+// Create creates a new, empty notmuch database located at 'path'.
+func Create(path string) (*DB, error) {
+	cpath := C.CString(path)
+	defer C.free(unsafe.Pointer(cpath))
+
+	db := &DB{}
+	cdb := (**C.notmuch_database_t)(unsafe.Pointer(&db.cptr))
+	cerr := C.notmuch_database_create(cpath, cdb)
+	runtime.SetFinalizer(db, func(db *DB) {
+		C.notmuch_database_destroy(db.toC())
+	})
+	return db, statusErr(cerr)
+}
+
+// Open opens the database at the location path using mode. Caller is responsible
+// for closing the database when done.
+func Open(path string, mode DBMode) (*DB, error) {
+	cpath := C.CString(path)
+	defer C.free(unsafe.Pointer(cpath))
+
+	cmode := C.notmuch_database_mode_t(mode)
+	db := &DB{}
+	cdb := (**C.notmuch_database_t)(unsafe.Pointer(&db.cptr))
+	cerr := C.notmuch_database_open(cpath, cmode, cdb)
+	runtime.SetFinalizer(db, func(db *DB) {
+		C.notmuch_database_destroy(db.toC())
```

Comment:
funny that it's always failing on Go 1.4 and passing on Go 1.5. @zenhack are we planning to support Go 1.4?